--- ./50_add_susfs_in_gki-android13-5.15.patch	2026-02-15 16:08:48.430199823 +0000
+++ ./50_add_susfs_in_gki-android13-5.15.patch.fixed	2026-02-15 16:09:19.841538095 +0000
@@ -697,9 +697,9 @@
 +#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
 +	copy_flags |= CL_COPY_MNT_NS;
 +#endif
- 	new = copy_tree(old, old->mnt.mnt_root, copy_flags);
- 	if (IS_ERR(new)) {
- 		namespace_unlock();
+ #ifdef CONFIG_KDP_NS
+	new = copy_tree(old, ((struct kdp_mount *)old)->mnt->mnt_root, copy_flags);
+ #else
 @@ -4669,3 +4868,42 @@ const struct proc_ns_operations mntns_operations = {
  	.install	= mntns_install,
  	.owner		= mntns_owner,
